### ISSUE TITLE
[Steam] GameMode option not working

### ISSUE THREAD

**Issue by sonic2kk** [libretro/RetroArch#13410]
## Description

Under Power Management settings, the "GameMode" option does not work. It is also set to **OFF**, even if the game is launched with GameMode from Steam. Plasma also mentions that GameMode is running.

### Expected behavior

RetroArch should allow GameMode to be enabled, and detect when it is launched with GameMode and update the menu option to **ON**.

### Actual behavior

RetroArch does not allow enabling GameMode, and does not update the menu option even if GameMode is running on the system when it is launched.

### Steps to reproduce the bug

(Optional) Add `gamemoderun %command%` to Launch Options for RetroArch
1. Launch RetroArch and go to Power Management settings
2. RetroArch will not allow enabling GameMode, throwing an error

### Version/Commit
- RetroArch: Steam/Nightly

### Environment information

- OS: Arch Linux w/ KDE Plasma 5.23.4 (Wayland), Mesa 21.3.2, Kernel 5.15.10

**Comment by RobLoach** [comment]
Do the logs indicate anything? There are some debug infos there.

**Comment by sonic2kk** [comment]
I enabled logging in settings to get a file in `logs/`, let me know if any additional logging is needed!

`[WARN] [GameMode]: GameMode cannot be enabled on this system ("dlopen failed - libgamemode.so: cannot open shared object file: No such file or directory.") https://github.com/FeralInteractive/gamemode needs to be installed.`

I can see the files for GameMode in `/usr/lib` on my system, including the specific library it's talking about. It appears to be a link to `libgamemode.so.0.0.0`, though it hasn't caused problems with other applications before.

Is this maybe an issue related to the Steam Linux Runtime with the Steam version of RetroArch?

**Comment by Augusto7743** [comment]
Here using Lubuntu 20.04.3 and installed gamemode 1.5.1.
Using Retroarch default current 64 bits version.
Gamemode is working correctly.

**Comment by RobLoach** [comment]
Could be that the Steam version doesn't read from the system binaries? We don't distribute GameMode.

**Comment by sonic2kk** [comment]
Indeed, the Steam version should use the Steam Linux Runtime, which I believe means it does not (cannot?) read from any system libraries. Someone with more experience than me might be able to shed more light though 😅

**Comment by hizzlekizzle** [comment]
I know we had talked about bundling it, since it's BSD 3-clause licensed. I think that's the real solution here.

**Comment by Radon86222** [comment]
I was having this same problem, but managed to fix it with the flatpak release of RetroArch by copying libgamemode.so (not the symbolic link, the actual file) out of the regular lib folder, then prefixing the command to launch flatpak with a LD_PRELOAD variable pointing to the new file. Appears to just be a permissions issue with the folder itself, rather than the library. For some reason, ld.so just can't access it while its in that folder, even if you give it all permissions.

Not sure if this solution is specific to the flatpak release or not.

**Comment by Radon86222** [comment]
> Use Flatseal to give RetroArch access to "All system libraries, executables and static data".

Unfortunately, this doesn't seem to work for me. Even specifically naming out the absolute path to where libgamemode.so is in Flatseal doesn't work.

**Comment by Theoldnation** [comment]
> I was having this same problem, but managed to fix it with the flatpak release of RetroArch by copying libgamemode.so (not the symbolic link, the actual file) out of the regular lib folder, then prefixing the command to launch flatpak with a LD_PRELOAD variable pointing to the new file. Appears to just be a permissions issue with the folder itself, rather than the library. For some reason, ld.so just can't access it while its in that folder, even if you give it all permissions.
> 
> Not sure if this solution is specific to the flatpak release or not.

i'm getting '/home/theoldnation/test/libgamemode.so"' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored. when trying to prefix and won't work at all both using flatseal and via the terminal

**Comment by jojorne** [comment]
That is correct! Steam cannot read from the system library. Use this to force steam to load it:
`LD_PRELOAD="$LD_PRELOAD:/usr/\$LIB/libgamemode.so" %command%`
